packages/gcp/data_stream/audit/elasticsearch/ingest_pipeline/default.yml

```
@@ -105,6 +105,118 @@ processors:
         API_VERSION: (v\d+([a-z]+)?(\d+)?)
         RESOURCE_TYPE: ([a-z]+((\.[a-z0-9]+)+)?)
       ignore_missing: true
+  - script:
+      description: Appends any relevant entity to `related.entity` for all events
+      lang: painless
+      on_failure:
+        - set:
+            description: Adds error reason to the document
+            field: error.message
+            value: "{{{ _ingest.on_failure_message }}}"
+      source: |
+        void addFields(Set entities, String[] fields) {
+            for (String field : fields) {
+              addField(entities, field);
+            }
+        }
+
+        void addField(Set entities, String fieldName) {
+          addValue(entities, field(fieldName).get(null));
+        }
+
+        boolean addValues(Set entities, String[] values) {
+            boolean addedAll = true;
+            for (String value : values) {
+              addedAll = addedAll && addValue(entities, value);
+            }
+
+            return addedAll;
+        }
+
+        boolean addValue(Set entities, String value) {
+          if (value == null || value == "") {
+            return false;
+          }
+
+         return entities.add(value);
```

```suggestion
          return entities.add(value);
```
Though same query here; while the `addValues` caller uses the return, no other caller does, and `addValues`'s return value is never used, so this seems unnecessary. I would not be surprised if all of this could be elided and done inline more concisely when the return values are removed.